teammapper-backend/test/db.ts

```diff
@@ -1,20 +1,78 @@
 import { TypeOrmModuleOptions } from '@nestjs/typeorm';
+import { DataSource } from 'typeorm';
 
 require('dotenv').config();
 
-export const createTestConfiguration = (): TypeOrmModuleOptions => ({
-  type: 'postgres',
-  host: process.env.POSTGRES_TEST_HOST,
-  port: parseInt(process.env.POSTGRES_TEST_PORT),
-  username: process.env.POSTGRES_TEST_USER,
-  password: process.env.POSTGRES_TEST_PASSWORD,
-  database: process.env.POSTGRES_TEST_DATABASE,
-  synchronize: true,
-  autoLoadEntities: true,
-  dropSchema: true,
-  keepConnectionAlive: true,
-  extra: {
-    query_timeout: 1000,
-    statement_timeout: 1000,
-  },
-});
+// we create a database for each worker:
+const setupWorkerDatabase = async (workerId: string): Promise<string> => {
+  const connection = await mainTestDataSource();
+  const databaseName = buildDatabaseName(workerId);
+  await connection.query(`DROP DATABASE IF EXISTS "${databaseName}"`);
+  await connection.query(`CREATE DATABASE "${databaseName}"`);
+  await connection.destroy();
+
+  return databaseName;
+}
+
+// this is the configuration for the main test database. this database is not used for actual tests.
+// it is the entrypoint for each worker to be able to create their own worker database
+const mainTestDataSource = async () => {
```

Comment:
i understand that there seem to be some differences between these two test configs (e.g. sync). Still I would propose not copy paste it here, but merge it - e.g.:

```
const connection = new DataSource({ ... workerDataSourceConfig, synchronize: false })
``` 

> we don't need to synchronise the tables for the main test database

Also i am still not 100% sure why now workers (`workerDataSourceConfig`) use synchronize true, but the main config sets this to false. Also you would need then to run migrations if this is set to false, no?

Reply:
Yes, I'll extract that to a function. The reason we don't run migrations in the main test database is, that no tests are running against this database. This database is just required to be able to initially connect to postgres, to then create additional databases. You cannot connect to postgres without a database.